generated/nidcpower/nidcpower/session.py

```diff
@@ -5156,6 +5163,169 @@ def create_advanced_sequence(self, sequence_name, property_names, set_as_active_
+                BoardType:<bus connector> & ResourceName:<resource name>; Model:<model number>;
+                BoardType:<bus connector>
+
+                You do not have to specify a value for all the properties. If you do not specify a
+                value for a property, the default value is used.
+
+            independent_channels (bool): Specifies whether to initialize the session with
+                independent channels. Set this argument to False on legacy applications or if you
+                are unable to upgrade your NI-DCPower driver runtime.
+
+
+        Returns:
+            vi (int): Returns a session handle that you use to identify the device in all
+                subsequent NI-DCPower method calls.
+
+        '''
+        if independent_channels:
+            resource_string = resource_name  # store potential modifications to resource_name in a separate variable
+
+            if channels:
+                # if we have a channels arg, we need to try and combine it with the resource name
+                # before calling into initialize with independent channels
+                channel_list = (f"{resource_name}/{channel}" for channel in channels.split(","))
+                resource_string = ",".join(channel_list)
+
+                import warnings
+                warnings.warn(
+                    "Attempting to initialize an independent channels session with a channels argument. The resource "
+                    f"name '{resource_name}' will be combined with the channels '{channels}' to form the "
+                    f"fully-qualified channel list '{resource_string}'. To avoid this warning, use a fully-qualified "
+                    "channel list as the resource name instead of providing a channels argument.",
+                    DeprecationWarning
+                )
+
+                if "," in resource_name:
+                    raise ValueError(
+                        f"Channels can't be combined with multiple devices in the resource name '{resource_name}'. Use "
+                        "a single device in the resource name or provide a list of fully-qualified channels as the "
+                        "resource name instead of supplying a channels argument."
+                    )
+
+            # TODO(smooresni): Modify this line as described in https://github.com/ni/nimi-python/issues/1596
+            self._vi = self._initialize_with_independent_channels(resource_string, reset, option_string)
+
+        else:
+            import warnings
+            warnings.warn("Initializing session without independent channels enabled.", DeprecationWarning)
+
+            # TODO(smooresni): Modify this line as described in https://github.com/ni/nimi-python/issues/1596
+            self._vi = self._initialize_with_channels(resource_name, channels, reset, option_string)
+
+        # TODO(smooresni): Remove the following code as described in https://github.com/ni/nimi-python/issues/1596
```

Comment:
The changes related to locking seems complex enough to warrant a separate PR. Can you please move it to a separate PR and have this PR add support for independent channels without any locking?

FYI @marcoskirsch

Reply:
Sure, I could do that. However, most methods apply the 'ivi_synchronized' decorator which calls into 'lock' and 'unlock'. Doing so will put nidcpower temporarily in a bad state as it will only work if 'independent_channels' is explicitly set to False. If that's okay in the meantime, I'll go ahead and split the PRs.

Reply:
That's fine.

Reply:
We can temporarily remove locking:
https://github.com/ni/nimi-python/blob/master/src/nise/metadata/config.py#L41

That way we don't push something completely broken.